web/src/api/api.ts

```diff
@@ -86,4 +86,35 @@ export class ImmichApi {
 	}
 }
 
+export type MapMarkerResponseDto = [
+	// assetId
+	string,
+	// latitude
+	number,
+	// longitude
+	number
+];
+
+export class MapMarkerResponseDtoDecorator {
+	private dto: MapMarkerResponseDto;
+
+	constructor(dto: MapMarkerResponseDto) {
+		this.dto = dto;
+	}
+
+	static map(response: object[]): MapMarkerResponseDtoDecorator[] {
```

Comment:
If you're going to have a class that represents this and take the time to convert from the array format into an instance for each record, you should just have named properties on this like private string assetId, etc. and they should be set in the constructor. Definitely should not be using unknown casting anywhere here.

Reply:
The problem is that OpenAPI generates `Array<object>` here. I don't think there is a way of casting that to an array without `unknown`?